File: Release/WetiMatlabFunctions/PreProcessing/GetSimulationName.m
%% GetSimulationName
% Function: Provides a standard Simulation Name from Variation
%
%% Usage:
% 
% SimulationName = GetSimulationName(PreProcessingVariation,Variation)
% or 
% SimulationName = GetSimulationName(PreProcessingVariation,Variation,Format)
%
%% Input:
% 
% * PreProcessingVariation  - Cell with String of Variation in first column
% * Variation               - Vector with floats of current Variation
% * Format (optional)       - Cell with format strings
% 
%% Output:
% GetSimulationName         - String
% 
%% Created: 
% David Schlipf on 26-Apr-2017
% 
% (c) TTI GmbH

function SimulationName = GetSimulationName(PreProcessingVariation,VariationValues)

% inputs
arguments
    PreProcessingVariation  (:,3) cell
    VariationValues         double
end

% loop
[nVariation,~]      = size(PreProcessingVariation);
SimulationNameTemp  = [];
for iVariation = 1:nVariation
    SimulationNameTemp = [SimulationNameTemp,...
        PreProcessingVariation{iVariation,1},'_'];
    Format = PreProcessingVariation{iVariation,3};
    if ~isempty(Format)
        SimulationNameTemp = [SimulationNameTemp,num2str(VariationValues(iVariation),Format),'_'];
    else
        SimulationNameTemp = [SimulationNameTemp,num2str(VariationValues(iVariation)),'_'];
    end
end

% remove last '_'
SimulationNameTemp(end) = [];

% replace special characters
SimulationName          = regexprep(SimulationNameTemp, {'\.','\-','\+'}, {'d','m','p'});

end

File: Release/PythonFunctions/PreProcessing/GetSimulationName.py
import numpy as np
import re

def GetSimulationName(pre_processing_variation, variation_values):
    n_variation = len(pre_processing_variation)
    simulation_name_temp = ""
    for i_variation in range(n_variation):
        simulation_name_temp += pre_processing_variation[i_variation][0] + "_"
        format_str = pre_processing_variation[i_variation][2]
        if format_str:
            simulation_name_temp += f'{int(variation_values[i_variation]):{format_str}}' + "_"
        else:
            simulation_name_temp += str(variation_values[i_variation]) + "_"

    # remove last '_'
    simulation_name_temp = simulation_name_temp[:-1]

    # replace special characters
    simulation_name = re.sub(r'[.\-+]', lambda x: {'.' : 'd', '-' : 'm', '+' : 'p'}[x.group()], simulation_name_temp)

    return simulation_name
